aiidalab_widgets_base/computers.py

```diff
@@ -0,0 +1,718 @@
+        self._btn_get_config = ipw.Button(description="Get from config",
+                                          layout={'margin': '20px 0px 20px 130px'},
+                                          disabled=True)
+        self._btn_get_config.on_click(self._get_from_config)
+
+        # Computer ssh settings
+        self._inp_computer_hostname = ipw.Text(description="Computer name:",
+                                     layout=ipw.Layout(width="350px"),
+                                     style=style)
+        self._inp_computer_hostname.observe(self.on_computer_hostname_change, names='value')
+        self._use_proxy = ipw.Checkbox(value=False, description='Use proxy')
+        self._use_proxy.observe(self.on_use_proxy_change, names='value')
+        computer_ssh_box = ipw.VBox([self._btn_get_config,
+                                     self._inp_username,
+                                     self._inp_password,
+                                     self._use_proxy])
+
+        # Proxy ssh settings
+        self._inp_proxy_address = ipw.Text(description="Proxy server address:",
+                                           layout=ipw.Layout(width="350px"),
+                                           style=style)
+        self._use_diff_proxy_username = ipw.Checkbox(value=False,
+                                               description='Use different username and password',
+                                               layout={'width': 'initial'})
+        self._use_diff_proxy_username.observe(self.on_use_diff_proxy_username_change, names='value')
+
+        self._inp_proxy_username = ipw.Text(value='',
+                                            description="Proxy server username:",
+                                            layout=ipw.Layout(width="350px"), style=style)
+        self._inp_proxy_password = ipw.Password(value='',
+                                                description="Proxy server password:",
+                                                layout=ipw.Layout(width="350px"),
+                                                style=style)
+        self._proxy_user_password_box = ipw.VBox([self._inp_proxy_username,
+                                                  self._inp_proxy_password],
+                                                 layout={'visibility':'hidden'})
+        self._proxy_ssh_box = ipw.VBox([self._inp_proxy_address,
+                                        self._use_diff_proxy_username,
+                                        self._proxy_user_password_box],
+                                 layout = {'visibility':'hidden'})
+
+        self._btn_setup_ssh = ipw.Button(description="Setup ssh")
+        self._btn_setup_ssh.on_click(self.on_setup_ssh)
+        self._setup_ssh_out = ipw.Output()
+
+        # Check if some settings were already provided
+        self._predefine_settings(**kwargs)
+        children = [ipw.HBox([computer_image, ipw.VBox([self._inp_computer_hostname,
```

Comment:
For me, the `computer_ssh_box` and `_proxy_ssh_box` cannot be fully shown in the width, and is scrollable.

I think this can be mediated by changing margins and pixel-widths

Reply:
fixed

Reply:
It should be possible to keep the password textbox in the same size as the username textbox, since the `Password` widget inherits from `Text`, but if it's too difficult/fiddly, it is also okay as it is now. It's purely aesthetic in the end.

Reply:
Well, of course if you know a way to fix that - I am happy to add your fix.

I tried to play with margins, width parameters. This did not result in any success.

To my understanding the only parameter that Password widget takes into account is `width="138px"` which affects the description width only.